cmake/configs/base.cmake

```diff
@@ -170,6 +170,12 @@ config_bool(
     DEFAULT ON
 )
 
+config_bool(
+    ENABLE_COLORIZE_OUTPUT
+    "Compile with build error colors enabled"
+    DEFAULT ON
```

Comment:
Is there a way to detect if stdout is a tty here? This is an improvement but it would be nice if I didn't have to add another flag to my "usual" list of defaults, and it just "did the right thing" without users having to request colours.

Reply:
I looked into using `isatty` using a shell script or something like `$ if [ -t 1 ] ; then echo terminal; else echo "not a terminal"; fi` (ty @ershov for the suggestion) but I'm worried it'll bring more issues than it fixes to try do this.
I've looked into a few resources:
- [src1](https://stackoverflow.com/questions/44558396/determine-whether-cmake-is-executing-from-terminal-or-gui): cant determine if cmake is executing from terminal or gui
- [src2](https://gitlab.kitware.com/cmake/cmake/-/issues/17143): open issue on cmake not detecting if tty is used or ANSI color escapes supported

For now it'll be better to stick with this. You wouldn't need to add another flag to your list of defaults unless you wanted to disable it.

Reply:
OK, that makes sense. Thanks for digging into this, it sounds like more effort than it's worth. My hope was that it would be easy.